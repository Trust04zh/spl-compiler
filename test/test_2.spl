int nya(int times) {
  int i = 1;
  while (1) {
    print();
    if (i <= times) {} else {break;}
  }
  return 0;
}