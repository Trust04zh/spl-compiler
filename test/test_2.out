Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: nya
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: times
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: int
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: i
                ASSIGN
                Exp (2)
                  INT: 1
            SEMI
        StmtList (3)
          Stmt (3)
            WHILE
            LP
            Exp (3)
              INT: 1
            RP
            Stmt (3)
              CompSt (3)
                LC
                StmtList (4)
                  Stmt (4)
                    Exp (4)
                      ID: print
                      LP
                      RP
                    SEMI
                  StmtList (5)
                    Stmt (5)
                      IF
                      LP
                      Exp (5)
                        Exp (5)
                          ID: i
                        LE
                        Exp (5)
                          ID: times
                      RP
                      Stmt (5)
                        CompSt (5)
                          LC
                          RC
                      ELSE
                      Stmt (5)
                        CompSt (5)
                          LC
                          StmtList (5)
                            Stmt (5)
                              Exp (5)
                                ID: break
                              SEMI
                          RC
                RC
          StmtList (7)
            Stmt (7)
              RETURN
              Exp (7)
                INT: 0
              SEMI
        RC
